### PR TITLE
Alistair/device pass thru

### PR DESCRIPTION
**Pull request by alistair23**
### Pull Request Overview

This is the complete implementation (not just HIL) of device pass through support, as mentioned in https://github.com/tock/tock/issues/4020

This replaces the original HIL PR from https://github.com/tock/tock/pull/4044. Hopefully a more complete implementation shows how this can work.

There are a few `todo!()`s in the capsule still, so I'm marking this as draft. But otherwise it's pretty ready to go.

### Testing Strategy

Running a BLE application

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make prepush`.

### CODE REVIEW

File: kernel/src/passthrough.rs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    fn command(
+        &self,
+        command_number: usize,
+        address: usize,
+        size: usize,
+        processid: ProcessId,
+    ) -> CommandReturn {
+        match command_number {
+            0 => CommandReturn::success(),
+            // Query memory information?
```
Comment (twilfredo) [inline kernel/src/passthrough.rs:74]: What type of information could we return here?

**Comment by hudson-ayers** [comment]
We talked about this for awhile on the call today, and I don't think there is high-level agreement that this is a mode of operation we want to bless by including it in the core kernel. It is fundamentally unsafe and at odds with Tock's architecture. Even as an optional component that a board has to opt into, we still believe that this suggests a level of general support for this approach that we do not wish to imply.

That said, this is a well motivated use-case. We believe that this would be easier to accept if:
- The HIL and passthrough and platform-filter code lived in `boards/` , not `kernel/`
- the implementation used the existing `add_mpu_region()` API on `Process` instead of adding a new one referencing device passthrough.
- Similarly, it should be feasible to keep the pass-through logic for interrupts out of `chips/` , likely by hooking one of `service_interrupt` or `handle_interrupt`
- We do have a TRD type “Experimental” that seems like a good fit in this case if it is worth documenting this use case and potential designs. But, we do want to be up front that a full, experimental TRD doesn’t imply we will upstream the support.


In summary, we would love to see an implementation of this that works without modifications to `kernel/`, and ideally without any modifications (especially memory unsafe modifications) to `chips/`.

**Comment by lschuermann** [comment]
It seems like this has major implications on Tock's threat model that we'd need to figure out before we'd want merge anything like this.
